xDripG5/Transmitter.swift

```diff
@@ -130,13 +141,46 @@ public final class Transmitter: BluetoothManagerDelegate {
 
                         self.log.info("Bonding request sent. Waiting user to respond.")
                     }
-                    let (timeMessage, activationDate, glucoseMessage) = try peripheral.control(shouldWaitForBond: status.bonded != 0x1)
+
+                    try peripheral.enableNotify(shouldWaitForBond: status.bonded != 0x1)
+                    defer {
+                        peripheral.disconnect()
+                    }
+
+                    let timeMessage = try peripheral.readTimeMessage()
+
+                    let activationDate = Date(timeIntervalSinceNow: -TimeInterval(timeMessage.currentTime))
+
+                    while let command: Command = {
+                        var c: Command?
+                        self.delegateQueue.sync {
+                            c = self.commandSource?.dequeuePendingCommand(for: self)
+                        }
+                        return c
+                    }() {
+                        do {
+                            _ = try peripheral.sendCommand(command, activationDate: activationDate)
+                            self.delegateQueue.async {
```

Comment:
The function of the `didComplete` and `didFail` methods is to allow the commander to make an immediate decision about how to handle subsequent commands, e.g. it might decide to not continue if an earlier command fails.

This needs to be **synchronous**. No need to dispatch.

Reply:
Thanks. I hope I've understood correctly. In the latest commit I'm not calling any of the `TransmitterCommandSource` from the delegate queue.